lnwire/onion_error.go

```diff
@@ -330,42 +330,43 @@ func (f FailIncorrectPaymentAmount) Error() string {
 // gross overpayment.
 //
 // NOTE: May only be returned by the final node in the path.
-type FailUnknownPaymentHash struct {
+type FailIncorrectOrUnknownPaymentDetails struct {
 	// amount is the value of the extended HTLC.
 	amount MilliSatoshi
 }
 
-// NewFailUnknownPaymentHash makes a new instance of the FailUnknownPaymentHash
-// error bound to the specified HTLC amount.
-func NewFailUnknownPaymentHash(amt MilliSatoshi) *FailUnknownPaymentHash {
-	return &FailUnknownPaymentHash{
+// NewFailIncorrectOrUnknownPaymentDetails makes a new instance of the
+// FailIncorrectOrUnknownPaymentDetails error bound to the specified HTLC
+// amount.
+func NewFailIncorrectOrUnknownPaymentDetails(amt MilliSatoshi) *FailIncorrectOrUnknownPaymentDetails {
+	return &FailIncorrectOrUnknownPaymentDetails{
 		amount: amt,
 	}
 }
 
 // Amount is the value of the extended HTLC.
-func (f FailUnknownPaymentHash) Amount() MilliSatoshi {
+func (f FailIncorrectOrUnknownPaymentDetails) Amount() MilliSatoshi {
 	return f.amount
 }
 
 // Code returns the failure unique code.
 //
 // NOTE: Part of the FailureMessage interface.
-func (f FailUnknownPaymentHash) Code() FailCode {
+func (f FailIncorrectOrUnknownPaymentDetails) Code() FailCode {
 	return CodeUnknownPaymentHash
 }
 
 // Returns a human readable string describing the target FailureMessage.
 //
 // NOTE: Implements the error interface.
-func (f FailUnknownPaymentHash) Error() string {
+func (f FailIncorrectOrUnknownPaymentDetails) Error() string {
```

Comment:
should these also be defined over pointer receivers as in #3188 ?

Reply:
yes, fixed